utilities/unittests/test_pytest_utils.py

```
@@ -2536,3 +2539,287 @@ def test_no_deselection_when_no_multiarch_tests(self):
+        testcases = testsuite.findall("testcase")
+        assert len(testcases) == 4, f"Expected 4 testcases, got {len(testcases)}"
+        synthetic = testcases[-1]
+        assert synthetic.get("classname") == "pytest_exit"
+        assert synthetic.get("name") == "storage_class_failure"
+        assert testsuite.get("errors") == "1"
+        assert testsuite.get("tests") == "4"
+        assert testsuite.get("failures") == "1"
+
+    @patch("utilities.pytest_utils.pytest.exit")
+    @patch("utilities.pytest_utils.get_data_collector_base_directory")
+    def test_exit_pytest_execution_stores_failure_info(self, mock_get_base_dir, mock_pytest_exit):
```

Ruff (0.15.21)

[warning] Unused method argument: `mock_pytest_exit`

(ARG002)